### PR TITLE
Split export results into pages #6

### PR DESCRIPTION
**Pull request by vrodriguezf**
#6

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -23,3 +23,6 @@ CHAT_EXPORT_PUBLIC_CHANNELS=False
 CHAT_EXPORT_PUBLIC_GROUPS=False
 CHAT_EXPORT_PRIVATE_CHANNELS=False
 CHAT_EXPORT_PRIVATE_GROUPS=False
+
+MAX_FILE_SIZE=0 # Bytes. Negative or zero for unlimited size of the results
```
Comment (mo1ein) [inline .env.example:27]: I think `JSON_FILE_PAGE_SIZE` is better.

Comment (mo1ein) [inline .env.example:27]: I think `JSON_FILE_PAGE_SIZE` is better.

File: bot.py [hunk 2]
```diff
@@ -862,6 +862,29 @@ def to_html():
     pass
 
 
+def split_json_file(data: dict, output_path: str, max_size: int = MAX_FILE_SIZE):
```
Comment (mo1ein) [inline bot.py:865]: `page_size` is better than `max_size`.

File: bot.py [hunk 3]
```diff
@@ -862,6 +862,29 @@ def to_html():
...
+    messages = data["messages"]
+    part_count = 1
```
Comment (mo1ein) [inline bot.py:867]: `page` is short and enough.

File: bot.py [hunk 4]
```diff
@@ -862,6 +862,29 @@ def to_html():
...
+    current_data = {"messages": []}
+
+    for msg in messages:
+        current_data["messages"].append(msg)
+        serialized = json.dumps(current_data, default=str)
```
Comment (mo1ein) [inline bot.py:872]: `serialized` is not clear. serialized what? I suggest use `serialized_pages` or sth like that.

File: bot.py [hunk 5]
```diff
@@ -862,6 +862,29 @@ def to_html():
...
+        if len(serialized.encode('utf-8')) > max_size:
+            # Remove the last message that made it too big
+            current_data["messages"].pop()
+            # Save current part
+            with open(output_path.replace("result.json", f"result_part{part_count}.json"), "w") as outf:
```
Comment (mo1ein) [inline bot.py:877]: Also should change here too `f"result_part{page}`.

File: bot.py [hunk 6]
```diff
@@ -890,8 +913,11 @@ async def main():
                 await archive.process_message(chat, message, msg_info)
 
             json_name = generate_json_name(username=archive.username)
-            with open(json_name, mode='w') as f:
-                json.dump(archive.chat_data, f, indent=4, default=str)
+            if MAX_FILE_SIZE <= 0:
```
Comment (mo1ein) [inline bot.py:916]: To improve clarity, I suggest we change the code to:
```
if not JSON_FILE_PAGE_SIZE:
    with open(json_name, mode='w') as f:
         json.dump(archive.chat_data, f, indent=4, default=str)
else:
    split_json_file(archive.chat_data, json_name, JSON_FILE_PAGE_SIZE)
```

File: configs.py [hunk 7]
```diff
@@ -35,5 +35,8 @@
     'private_groups': env.get("CHAT_EXPORT_PRIVATE_GROUPS").lower() in ('true', '1', 't'),
 }
 
+# Negative value means no splitting
+MAX_FILE_SIZE = int(env.get("MAX_FILE_SIZE"))
```
Comment (mo1ein) [inline configs.py:39]:
```
JSON_FILE_PAGE_SIZE = None if env.get('JSON_FILE_PAGE_SIZE') == '' else int(env.get('JSON_FILE_PAGE_SIZE'))
```

**Review by mo1ein — COMMENTED** [review]
Just a few tiny changes are needed. Thanks, Victor!

File: bot.py (hunk 5 above)
Comment (vrodriguezf) [reply bot.py:877]: not sure if I understand this

File: .env.example [hunk 8]
```diff
@@ -23,3 +23,6 @@ CHAT_EXPORT_PUBLIC_CHANNELS=False
 CHAT_EXPORT_PUBLIC_GROUPS=False
 CHAT_EXPORT_PRIVATE_CHANNELS=False
 CHAT_EXPORT_PRIVATE_GROUPS=False
+
+JSON_FILE_PAGE_SIZE=0 # Bytes. Negative or zero for unlimited size of the results
```
Comment (mo1ein) [inline .env.example:27]: Change comment to: 
```suggestion
# Bytes. None for unlimited size of the results.
```

**Comment by mo1ein** [comment]
LGTM! Merged. 🚀

**Pull request merged**